CDP4ShellDialogs/ViewModels/OpenIdAuthenticationDialogViewModel.cs

```diff
@@ -0,0 +1,206 @@
+//    version 3 of the License, or any later version.
+// 
+//    The CDP4-COMET-IME Community Edition is distributed in the hope that it will be useful,
+//    but WITHOUT ANY WARRANTY; without even the implied warranty of
+//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//    GNU Affero General Public License for more details.
+// 
+//    You should have received a copy of the GNU Affero General Public License
+//    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace CDP4ShellDialogs.ViewModels
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net;
+    using System.Net.Http;
+    using System.Net.NetworkInformation;
+    using System.Text.Json;
+    using System.Threading.Tasks;
+    using System.Web;
+
+    using CDP4Composition.Navigation;
+
+    using CDP4ShellDialogs.Model;
+
+    /// <summary>
+    /// The <see cref="OpenIdAuthenticationDialogViewModel" /> is a <see cref="DialogViewModelBase" /> that provides OpenID authentication support
+    /// </summary>
+    public class OpenIdAuthenticationDialogViewModel : DialogViewModelBase
+    {
+        /// <summary>
+        /// Gets the <see cref="System.Text.Json.JsonSerializerOptions" />
+        /// </summary>
+        private static readonly JsonSerializerOptions JsonSerializerOptions = new JsonSerializerOptions
+        {
+            PropertyNamingPolicy = JsonNamingPolicy.SnakeCaseLower
+        };
+
+        /// <summary>
+        /// Gets the authority uri
+        /// </summary>
+        private readonly string authority;
+
+        /// <summary>
+        /// Gets the callback uri
+        /// </summary>
+        private readonly string callbackUri;
+
+        /// <summary>
+        /// Gets the identifier of the client
+        /// </summary>
+        private readonly string clientId;
+
+        /// <summary>
+        /// Gets the <see cref="HttpListener" />
+        /// </summary>
+        private readonly HttpListener listener;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OpenIdAuthenticationDialogViewModel" />
+        /// </summary>
+        /// <param name="authority">The URI to the OpenID authority</param>
+        /// <param name="clientId">The identifier of the specific client that provides authentication</param>
+        public OpenIdAuthenticationDialogViewModel(string authority, string clientId)
+        {
+            var port = 0;
+
+            if (!TryGetUnusedPort(49215, ref port))
+            {
+                throw new InvalidOperationException("Unable to get unused port");
+            }
+
+            this.authority = authority.TrimEnd('/');
+            this.clientId = clientId;
+            this.callbackUri = $"http://127.0.0.1:{port}/";
```

Comment:
Should this be a fixed value of 127.0.0.1?

Reply:
Indeed, from the keycloak documentation; 
![image](https://github.com/user-attachments/assets/912c17b5-dad4-4117-9c9c-fc629fffa182)